draft-ion.adoc

```diff
@@ -208,79 +200,176 @@ represented as follows:
+
+If a member name is not equal to the case-sensitive octet sequence `meta`, an Ion parser MUST NOT identify a member as a _Meta Object_.
+
+The member names within an identified _Meta Object_ MUST be unique; Ion parsers MUST either
+reject _Meta Object_ members with duplicate member names or use a JSON
+parser that returns only the lexically last duplicate member, as specified
 in {uri-ecmaScript-5-1-15-12}[Section 15.12 (The JSON Object)] of {uri-ecmaScript-5-1}[ECMAScript 5.1].
 
-The set of fields that a _Meta Object_ must contain to be considered valid
+The set of members that a _Meta Object_ must contain to be considered valid
 is resource-dependent.  In the absense of any resource-specific requirements,
-any field not understood by implementations MUST be ignored.
+any member not understood by implementations MUST be ignored.
+
+[[links]]
+== Links
+
+An Ion Link is an Ion Object where its `meta` member contains an {uri-IRIs}[IRI] member named `href`. An Ion Link is therefore JSON serialization of a {uri-rfc5988}[web link] to another resource.
+
+Ion parsers MUST identify any JSON structure matching this definition as an Ion Link.  Anything that does not match this defined structure MUST NOT be considered an Ion Link.
+
+A non-normative example of a minimal Ion link:
+
+.Example simplest Ion link:
+[source,json]
+----
+{ "meta": { "href": "https://api.example.com/corporations/acme" } }
+----
+
+=== Link Relation Types
+
+All Ion links have one or more <<links, link relation types>>.  A link's total set of link relation types is the set union of the link's implicit link relation type and the link's explicit link relation types.
+
+==== Implicit Link Relation Type
+
+All Ion links have exactly one implicit link relation type.  The implicit link relation type can be one of three possible values depending on where the link is located in the Ion content:
+
+1. Root Object
+2. Named Link
+3. Collection Resource `items` array element
+
+===== Root Object Implicit Link Relation Type
+
+If the Root Object is also found to be an Ion Link, it has an implicit link relation type of `self`.
+
+A non-normative example of a Root Object that is also an Ion Link:
+
+[source,json]
+----
+{
+  "meta": { "href": "https://api.example.com/users/1" },
+  "name": "Joe"
+}
+----
 
-=== Registered Metadata Field Names
+===== Named Link Implicit Link Relation Type
 
-The following field names are registered in the IANA Ion Metadata field registry
-defined in [TBD].  None of the fields defined below are intended to be mandatory
+If a JSON Object member value is an Ion Link, the Ion Link is implicitly assigned a link relation type that equals the member name.
+
+A non-normative example:
+
+[source,json]
+----
+{
+  "employer": { "meta": { "href": "https://api.example.com/corporations/acme" } }
+}
+----
+
+The above example shows an Ion Link member named `employer`.  The octet sequence `employer` is therefore assigned as the link's implicit link relation type.
```

Comment:
I'm glad you guys are digging in to this, as yes, this one issue is perhaps the most important of the entire Ion spec.

So I have a scenario that hopefully illustrates my concerns and might elucidate ways to address this:

Let's say I have a web application and I use a shiny new Logging-as-a-Service product, I'll call it 'Loggit'.  Today, Loggit exposes a 'REST' API that allows me to POST a single log entry so it can aggregate it and allow me to search it later:

```json
{
    "2015-06-12T20:15:43Z": "some random log line here with lots of data"
}
```

I integrate w/ Loggit and deploy my application to production and all is great.

Tomorrow, Loggit decides that the structure isn't flexible enough, and that instead of a single date/value pair, log entries must be submitted as follows:

```json
{
    "date": "ISO-8601-value-here",
    "ip": "IPV4-or-IPV6-value-here",
    // other name/value pairs as necessary
}
```

If they do this, claiming 'evolvability' and loose coupling, how is my application supposed to react to this sort of change so that my app does not break?  No human being will be able to react to such change at the time it occurs.

In the spirit of this question, I should point out that the focus for me w/ Ion since the beginning has been about _machine_ data interchange, not human data interchange, hence the subtitle:

> A human-friendly JSON-based hypermedia type **for machine data interchange**.

But Ion content should be easy to understand - even pleasant to look at - for people too.  I think this means that document members should be intuitively named and not susceptible to dramatic structural changes.

Thoughts?